### PR TITLE
sensor: mcp9808: convert to use device tree

### PR DESCRIPTION
**Pull request by mniestroj**
Convert the MCP9808 driver to use device tree and DT_INST_* defines.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

### CODE REVIEW

File: dts/bindings/sensor/microchip,mcp9808.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    !include i2c-device.yaml
+
+properties:
+    compatible:
+      constraint: "microchip,mcp9808"
+
+    int-gpios:
+      type: compound
+      category: optional
+      generation: define
```
Comment (galak) [inline dts/bindings/sensor/microchip,mcp9808.yaml]: remove `generation`, we don't need it anymore.

**Review by galak — CHANGES_REQUESTED** [review]
Minor update to binding, but otherwise looks ok.

**Review by ulfalizer — APPROVED** [review]
Kconfig changes seem fine.

**Comment by mniestroj** [comment]
@galak I have removed `generation` from device-tree bindings, as requested.

File: samples/sensor/mcp9808/sample.yaml [hunk 2]
```diff
@@ -4,4 +4,4 @@ tests:
   sample.sensor.mcp9808:
     harness: sensor
     tags: sensors
-    depends_on: i2c
+    depends_on: i2c mcp9808
```
Comment (MaureenHelm) [inline samples/sensor/mcp9808/sample.yaml]: There aren't any boards that set `mcp9808` as supported, so this change will cause the sample to never build in CI. Can you either revert this change or add `mcp9808` to the supported list for an appropriate board?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: samples/sensor/mcp9808/sample.yaml (hunk 2 above)
Comment (mniestroj) [reply samples/sensor/mcp9808/sample.yaml]: I don't know any already supported board that has that sensor, so I'll revert that change, as suggested.

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@mniestroj Please address the shippable failure

**Comment by mniestroj** [comment]
@MaureenHelm I needed to add dts_fixup.h for mcp9808 sample. I haven't found anything similar for other sensor samples, however all sensors that are already converted to DT are part of some supported boards in Zephyr (which is not the case with mcp9808). Are you ok with that approach?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
> @MaureenHelm I needed to add dts_fixup.h for mcp9808 sample. I haven't found anything similar for other sensor samples, however all sensors that are already converted to DT are part of some supported boards in Zephyr (which is not the case with mcp9808). Are you ok with that approach?

No, in that case you should add a dts overlay to the sample.

**Comment by mniestroj** [comment]
> No, in that case you should add a dts overlay to the sample.

For whatever board I choose? Should I do that similar to `adt7420` sensor example?

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@galak please revisit

**Comment by mniestroj** [comment]
@galak Could you review updated code, so we can move forward with this PR?

**Comment by mniestroj** [comment]
This MR has been superseded by #20413 .

**Pull request closed without merging**